src/Sdk/WebApi/WebApi/Results/ResultsHttpClient.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.Services.Results.Contracts;
+using System.Net.Http.Formatting;
+using Sdk.WebApi.WebApi;
+
+namespace GitHub.Services.Results.Client
+{
+    public class ResultsHttpClient : RawHttpClientBase
```

Comment:
one thing to check if you plan to use `ResultsHttpClient` in production for customers, does the RawHttpClientBase support all the network settings for the runner, ex: proxy, SSL cert, etc?

Reply:
it should, but I didn't review the initial PR and am not sure whether we implement all required pieces, it was added to testing RunService only in the dev environment.

Reply:
From what I can tell, `RawHttpMessagehandler.cs` handles [timeouts](https://github.com/actions/runner/blob/main/src/Sdk/Common/Common/RawHttpMessageHandler.cs#L129), [retries](https://github.com/actions/runner/blob/main/src/Sdk/Common/Common/RawHttpMessageHandler.cs#L145) and has [partial support for proxy](https://github.com/actions/runner/blob/main/src/Sdk/Common/Common/RawHttpMessageHandler.cs#L83) - it doesn't apply proxy authentication.  

It also doesn't support certs which is okay in Results case since we don't need to support self signed certificate yet.  

I will try to add the proxy authentication support in the other issue created.